// reapp-dev-tools/lib/components/StyleguidePage.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.StyleguidePage = undefined;

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _class, _temp;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _Button = require('react-bootstrap/lib/Button');

var _Button2 = _interopRequireDefault(_Button);

var _Panel = require('react-bootstrap/lib/Panel');

var _Panel2 = _interopRequireDefault(_Panel);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var StyleguidePage = exports.StyleguidePage = (_temp = _class = function (_React$Component) {
    _inherits(StyleguidePage, _React$Component);

    function StyleguidePage() {
        _classCallCheck(this, StyleguidePage);

        return _possibleConstructorReturn(this, Object.getPrototypeOf(StyleguidePage).apply(this, arguments));
    }

    _createClass(StyleguidePage, [{
        key: 'render',
        value: function render() {
            var styleguideRoot = this.context.styleguideRoot;
            var _props = this.props;
            var name = _props.name;
            var component = _props.component;
            var info = _props.info;

            var componentName = name.replace('.guide', '');
            var componentFile = componentName;

            var componentPath;
            if (info.plugin) {
                componentPath = ['app', 'plugins', info.plugin, 'components', componentFile].join('/');
            } else {
                componentPath = ['app', 'client', 'components', componentFile].join('/');
            }

            var source = _react2.default.createElement('code', {
                style: {
                    background: 'transparent'
                },
                children: componentPath });

            var footer = _react2.default.createElement(
                'div',
                null,
                _react2.default.createElement(_Button2.default, {
                    href: '#top',
                    bsStyle: 'link',
                    bsSize: 'xsmall',
                    style: { marginTop: -3 },
                    className: 'pull-right',
                    children: 'Back to Top' }),
                _react2.default.createElement('small', { children: source })
            );

            var Component = _react2.default.createElement(component);

            return _react2.default.createElement(
                'div',
                { id: name },
                _react2.default.createElement(_Panel2.default, {
                    header: componentName,
                    footer: footer,
                    bsStyle: 'primary',
                    style: { marginTop: 10 },
                    children: Component })
            );
        }
    }]);

    return StyleguidePage;
}(_react2.default.Component), _class.contextTypes = {
    styleguideRoot: _react2.default.PropTypes.string
}, _class.propTypes = {
    name: _react2.default.PropTypes.string.isRequired,
    component: _react2.default.PropTypes.func.isRequired,
    info: _react2.default.PropTypes.object.isRequired
}, _temp);
