// reapp-dev-tools/lib/components/Styleguide.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.Styleguide = undefined;

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _class, _temp;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _Grid = require('react-bootstrap/lib/Grid');

var _Grid2 = _interopRequireDefault(_Grid);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var Styleguide = exports.Styleguide = (_temp = _class = function (_React$Component) {
    _inherits(Styleguide, _React$Component);

    function Styleguide() {
        _classCallCheck(this, Styleguide);

        return _possibleConstructorReturn(this, Object.getPrototypeOf(Styleguide).apply(this, arguments));
    }

    _createClass(Styleguide, [{
        key: 'getChildContext',
        value: function getChildContext() {
            return {
                styleguideSources: this.props.sources,
                styleguideRoot: this.props.root
            };
        }
    }, {
        key: 'render',
        value: function render() {
            var children = this.props.children;

            return _react2.default.createElement(
                _Grid2.default,
                null,
                children
            );
        }
    }]);

    return Styleguide;
}(_react2.default.Component), _class.propTypes = {
    sources: _react2.default.PropTypes.array,
    root: _react2.default.PropTypes.string,
    children: _react2.default.PropTypes.array
}, _class.childContextTypes = {
    styleguideSources: _react2.default.PropTypes.array,
    styleguideRoot: _react2.default.PropTypes.string
}, _temp);
