// reapp-dev-tools/lib/components/StyleguideSectionBody.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.StyleguideSectionBody = undefined;

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _class, _temp;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var StyleguideSectionBody = exports.StyleguideSectionBody = (_temp = _class = function (_React$Component) {
    _inherits(StyleguideSectionBody, _React$Component);

    function StyleguideSectionBody() {
        _classCallCheck(this, StyleguideSectionBody);

        return _possibleConstructorReturn(this, Object.getPrototypeOf(StyleguideSectionBody).apply(this, arguments));
    }

    _createClass(StyleguideSectionBody, [{
        key: 'render',
        value: function render() {
            var style = {
                border: '1px dotted #ddd',
                borderRadius: 3,
                padding: 10,
                boxShadow: '1px 1px 3px #eee'
            };
            return _react2.default.createElement(
                'div',
                { style: style },
                this.props.children
            );
        }
    }]);

    return StyleguideSectionBody;
}(_react2.default.Component), _class.propTypes = {
    children: _react2.default.PropTypes.oneOfType([_react2.default.PropTypes.element, _react2.default.PropTypes.array])
}, _temp);
